sos/plugins/ovirt.py

```diff
@@ -66,7 +67,7 @@ class Ovirt(Plugin, RedHatPlugin):
     ]
 
     def setup(self):
-        if self.get_option('jbosstrace'):
+        if self.get_option('jbosstrace') and 'ovirt-engine' in self.packages:
```

Comment:
Is this test correct? Given how the plugin is currently written, `'ovirt-engine` in self.packages` will always evaluate to `True` (it's in the hard-coded packages list in the class declaration - normally when you have things checking that list, it's because there are separate `RedHatPlugin` `UbuntuPlugin` etc. versions with variant package lists).

Reply:
I am guessing you actually wanted to do a test against `self.policy().package_manager`...

Reply:
This worked for me, but admittedly I only tested this part once. I'll need to rebuild a test system for this. I'll try and get that done today.

Reply:
Here is the problem. The `oVirt.packages` tuple is assigned exactly once in this file, near the top of the class definition, and containing the `"ovirt-engine"` string:

```python
# Class name must be the same as file name and method names must not change
class Ovirt(Plugin, RedHatPlugin):
    """oVirt Engine"""

    plugin_name = "ovirt"
    profiles = ('virt',)

    packages = (
        'ovirt-engine',
        'ovirt-engine-dwh',
        'ovirt-engine-reports',
        'ovirt-scheduler-proxy',
        'rhevm',
        'rhevm-dwh',
        'rhevm-reports'
    )
```

The `packages` member is never re-assigned or modified in this plugin. This means that the 2nd clause in the test:


```python
        if self.get_option('jbosstrace') and 'ovirt-engine' in self.packages:
```

Can never evaluate `False`. It is exactly equivalent to the line that it replaces. To check whether a package is *installed* (rather than a trigger for the current plugins), you either need to reach into the policy and package manager classes:

```python
        self.policy().package_manager.all_pkgs()['net-tools']
```

Which I admit is hideously ugly (patches welcome! ;-), or you can use the `Plugin.is_installed(package)` method:

```python
        if self.get_option('jbosstrace') and self.is_installed('ovirt-engine'):
```

Which looks even better and more readable than the original.

Reply:
*double checks local branches*
I'm an idiot. I had a different set of changes between my laptop and desktop, and apparently never actually committed from my laptop, which is using `is_installed` (and was the only place I was looking for this one).

Will fix this is about a minute.

Reply:
Np, that's what reviews are for :)

Thanks for fixing it up.